### PR TITLE
APP-11738 Search

### PR DESCRIPTION
**Pull request by webspaceadam**
## Summary

- [APP-11738](https://flipapp.myjetbrains.com/youtrack/issue/APP-11738)

### CODE REVIEW

File: apps/swirl-docs/src/components/Search/Algolia.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import algoliasearch from "algoliasearch/lite";
+import { InstantSearch } from "react-instantsearch-dom";
+import SearchBox from "./SearchBox";
+import Hits from "./Hits";
+import { ALGOLIA_INDEX } from "@swirl/lib/search";
+
+// Initialize the Algolia client
+const searchClient = algoliasearch(
+  process.env.NEXT_PUBLIC_ALGOLIA_APP_ID!!,
+  process.env.NEXT_PUBLIC_ALGOLIA_SEARCH_API_KEY!!
```
Comment (Sqrrl) [inline apps/swirl-docs/src/components/Search/Algolia.tsx:11]: **question:** Just to make sure. This is a public API key which is save to be used here?

File: apps/swirl-docs/src/components/Search/SearchBox.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// import connectSearchBox
+import { connectSearchBox } from "react-instantsearch-dom";
+
+function SearchBox({ refine }: any) {
+  return (
+    <>
+      <input
```
Comment (Sqrrl) [inline apps/swirl-docs/src/components/Search/SearchBox.tsx:7]: **issue:** Needs a label.

File: apps/swirl-docs/src/scripts/build-search.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  DOCUMENTATION_CATEGORY,
+  createStaticPathsData,
+  DocumentationCategory,
+  StaticPathMap,
+} from "@swirl/lib/docs";
+import { generateSerializableDocumentation } from "@swirl/lib/docs/src/singleDoc";
+import { ALGOLIA_INDEX } from "@swirl/lib/search";
+import algoliasearch from "algoliasearch";
+
+import dotenv from "dotenv";
```
Comment (Sqrrl) [inline apps/swirl-docs/src/scripts/build-search.ts:11]: **suggestion:** Could you add a `.env.sample` file to the repo, including all used keys with empty values? Makes it easier to set it up locally. Could also be part of the setup steps for local development in the README.

File: apps/swirl-docs/src/scripts/build-search.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+    if (!process.env.ALGOLIA_SEARCH_ADMIN_KEY) {
+      throw new Error("ALGOLIA_SEARCH_ADMIN_KEY is not defined");
+    }
+
+    const components = getAlgoliaDataForCategory(
+      DOCUMENTATION_CATEGORY.COMPONENTS
+    );
+    const tokens = getAlgoliaDataForCategory(DOCUMENTATION_CATEGORY.TOKENS);
+    // const icons = getAlgoliaDataForCategory(DOCUMENTATION_CATEGORY.ICONS);
```
Comment (Sqrrl) [inline apps/swirl-docs/src/scripts/build-search.ts:64]: **question:** Ist this intentional? No icon indexing?

Comment (webspaceadam) [reply apps/swirl-docs/src/scripts/build-search.ts:64]: Currently there is only one page. so the return of the function is empty :)

File: apps/swirl-docs/src/components/Search/Algolia.tsx (hunk 1 above)
Comment (webspaceadam) [reply apps/swirl-docs/src/components/Search/Algolia.tsx:11]: yes, based on the [tutorial](https://www.algolia.com/doc/guides/building-search-ui/getting-started/react/) at least :D

File: apps/swirl-docs/src/components/HeaderNavigation/index.tsx [hunk 5]
```diff
@@ -26,44 +27,45 @@ const HeaderNavigation = () => {
...
-              src="/swirl-icon-temp.svg"
-              width={32}
-              height={32}
-            />
-            <span className="font-bold ml-3">Swirl</span>
-          </a>
-        </Link>
+        <div className="inline-flex w-full justify-between md:justify-start">
+          <Link href="/">
+            <a className="flex justify-center items-center mr-8">
```
Comment (Sqrrl) [inline apps/swirl-docs/src/components/HeaderNavigation/index.tsx:34]: **question:** Is the inner `<a>` still needed with Next.js 13?

**Review by Sqrrl — CHANGES_REQUESTED** [review]
The search field width changes when typing.

https://user-images.githubusercontent.com/1019414/211552805-e62f838a-049c-465e-89ec-c08305c169cb.mov

File: apps/swirl-docs/src/components/HeaderNavigation/index.tsx (hunk 5 above)
Comment (webspaceadam) [reply apps/swirl-docs/src/components/HeaderNavigation/index.tsx:34]: Possibly not, but this project is still in v12 😅 i wanted to do the update after the initial release, or at least when the site is ready to launch. so i think after the improvement tickets and the initial search there is room for that :)

Comment (Sqrrl) [reply apps/swirl-docs/src/components/HeaderNavigation/index.tsx:34]: Sorry, assumed you were on 13.

**Review by Sqrrl — APPROVED** [review]

**Pull request merged**